From 827478c562f00007f7d0c814afcc54c14ed595e8 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:10:32 -0400
Subject: [powerpc] Don't build assembly files with ABIv2

Message-id: <1410545655-205645-404-git-send-email-dzickus@redhat.com>
Patchwork-id: 94464
O-Subject: [RHEL7 PATCH 403/626] powerpc: Don't build assembly files with ABIv2
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 58aedccb1907f05f702f0f6d8f8a57e8efe485b7
Author: Anton Blanchard <anton@samba.org>
Date:   Wed Apr 23 10:02:04 2014 +1000

    powerpc: Don't build assembly files with ABIv2

    We avoid ABIv2 when building c files since commit b2ca8c89 (powerpc:
    Don't use ELFv2 ABI to build the kernel). Do the same for assembly
    files.

    Signed-off-by: Anton Blanchard <anton@samba.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/Makefile b/arch/powerpc/Makefile
index c0b031b..81ca430 100644
--- a/arch/powerpc/Makefile
+++ b/arch/powerpc/Makefile
@@ -115,6 +115,7 @@ endif
 
 CFLAGS-$(CONFIG_PPC64) := -mtraceback=no -mcall-aixdesc
 CFLAGS-$(CONFIG_PPC64) += $(call cc-option,-mabi=elfv1)
+AFLAGS-$(CONFIG_PPC64) += $(call cc-option,-mabi=elfv1)
 CFLAGS-$(CONFIG_PPC64) += $(call cc-option,-mcmodel=medium,-mminimal-toc)
 CFLAGS-$(CONFIG_PPC64) += $(call cc-option,-mno-pointers-to-nested-functions)
 CFLAGS-$(CONFIG_PPC32) := -ffixed-r2 $(MULTIPLEWORD)
@@ -138,7 +139,7 @@ CFLAGS-$(CONFIG_E6500_CPU) += $(call cc-option,-mcpu=e6500,$(E5500_CPU))
 CFLAGS-$(CONFIG_TUNE_CELL) += $(call cc-option,-mtune=cell)
 
 KBUILD_CPPFLAGS += -Iarch/$(ARCH)
-KBUILD_AFLAGS += -Iarch/$(ARCH)
+KBUILD_AFLAGS += -Iarch/$(ARCH) $(AFLAGS-y)
 KBUILD_CFLAGS += -msoft-float -pipe -Iarch/$(ARCH) $(CFLAGS-y)
 CPP  = $(CC) -E $(KBUILD_CFLAGS)
 
-- 
1.7.1